docroot/modules/custom/yukon_sso/yukon_sso.module

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * @file
+ * Main module file.
+ */
+
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Implements hook_openid_connect_userinfo_save().
+ */
+function yukon_sso_openid_connect_userinfo_save($account, $context) {
+  // All roles should be updated on each authorization.
+  // Delete all user roles.
+  foreach ($account->getRoles() as $role) {
+    $account->removeRole($role);
```

Comment:
My preference would be to only remove roles that are enabled through SSO. E.g. only remove a role if it's in the list returned by `yukon_sso_get_roles_mapping`.

In D7, there are some supplementary roles that aren't managed through SSO. It would be nice if we could continue with that approach.

Reply:
I thought that all roles should be managed through SSO.
It is an important information. Thanks.